src/Components/Endpoints/src/Binding/FormDataConverterOfT.cs

```diff
@@ -0,0 +1,9 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Endpoints.Binding;
+
+internal abstract class FormDataConverter<T> : FormDataConverter
+{
+    internal abstract bool TryRead(ref FormDataReader context, Type type, FormDataMapperOptions options, out T? result, out bool found);
```

Comment:
Is `type` ever going to be different from `typeof(T)`?

Reply:
It's not obvious yet in this PR, but it's passed in for a different reason. When you do `typeof(T)` the generated code needs to ask the type system at runtime (this is in IL) for the type instance, which involves a lookup. For that reason, many serializers cache and pass the type instance along.

If you look at System.Text.Json, they do something similar. See [here](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1.read?view=net-7.0#system-text-json-serialization-jsonconverter-1-read(system-text-json-utf8jsonreader@-system-type-system-text-json-jsonserializeroptions))

The other part is that we might still need this if in the future we want to support polymorphic binding or enable people to do so, so we can't just assume typeof(T) == type in general.